### PR TITLE
Folds passed to `cv.glmnet`

### PR DESCRIPTION
**Pull request by rachaelvp**
- Fixed bug with `prediction_bounds` (a `fit_hal` argument in `fit_control` list), which would error when it was specified as a numeric vector. Also, added a check to assert this argument is correctly specified.
- Simplified `fit_control` list arguments in `fit_hal`. Users can still specify additional arguments to `cv.glmnet` and `glmnet` in this list.
- Defined `weights` argument in `fit_hal`, instead of it being an optional argument in `fit_control`, to facilitate specification and avoid confusion. This increases flexibility with `SuperLearner` wrapper `SL.hal9001`; `fit_control` can now be customized in `SL.hal9001`.

### CODE REVIEW

File: R/hal.R [hunk 1]
```diff
@@ -186,6 +187,7 @@ fit_hal <- function(X,
                     family = c("gaussian", "binomial", "poisson", "cox"),
                     lambda = NULL,
                     id = NULL,
```
Comment (nhejazi) [inline R/hal.R:189]: a little outside the scope of this PR, but i managed to get at least somewhat confused about the intended difference between `id` and `foldid` (in `fit_control`). in trying to understand this, i noticed that the only reference ever made to the `id` argument is in line 279/281, where it's passed to the `cluster_ids` argument of `make_folds`. i don' have a good understanding of why we would need both, e.g., could we not recommend that users simply generate the folds externally via a mechanism analogous to that already implemented here and then pass the corresponding folds via the `foldid` argument when working with repeated measures data? my concern here is about "constructor creep" (too many similar arguments) and a possible lack of clarity (+ headache) for end users.

**Review by nhejazi — CHANGES_REQUESTED** [review]
this is great --- thanks very much for the work in catching these issues @rachaelvp! some comments/changes to be addressed and then we should be good to merge this PR; happy to discuss any of these items as needed. also, after completing these changes and before merging, we should document the bugs and corresponding fixes in the `NEWS.md` file.

for reference, i think i now understand how the `weights` argument slipped off our radar. during the rehaul of the `fit_hal` constructor (at v0.3.0 and then again at v0.4.0), we implemented the `fit_control` argument scheme while also removing the mechanism for passing arbitrary arguments to `glmnet` / `cv.glmnet` via the `...` args (as was available in v0.2.8 of the package, cf, https://github.com/tlverse/hal9001/blob/3f0503ffb73508821c19ba57b3a2aabdcf854aba/R/hal.R#L112). coincidentally, we were (to some degree?) cognizant of this when you made necessary changes to `haldensify` to appropriately pass `weights` to `fit_hal` via the `fit_control` mechanism, as shown in https://github.com/nhejazi/haldensify/commit/0881589da979d0f0b55244758a1b1b45509c5131#diff-8f69ca3bd5222ef916b8f7d8df76490d7a49c159f4f9220edfb7951684be88f2R57, which i guess correctly implemented the fix in `haldensify` but not in `hal9001` 🧐

**Comment by nhejazi** [comment]
thanks also for pointing out the issue with `make site`, i'll look into this separately

**Comment by rachaelvp** [comment]
Hey @nhejazi thanks so much for your thorough review of this. I just realized that these changes are not urgent; they can be merged into devel. Even though `...` is not listed in `fit_hal`, the documentation for `fit_control` states that any arguments to `glmnet` or `cv.glmnet` can be passed in `fit_hal` via `fit_control` list. This is why were successfully able to pass `weights` in `fit_control` list with `haldensify` and `sl3`. 

I noticed that there is no information on version 0.4.3 in the NEWS file. Should something be added for version 0.4.3, such as what's in the NEWS for version 0.4.2?

**Comment by nhejazi** [comment]
thanks for clarifying @rachaelvp, that's very helpful --- the changes to the `fit_hal()` constructor happened fast enough that i'd lost track, but it's great to realize that it's not a severe bug. it may be worth discussing adding a check to the `fit_control` arguments (if we don't already) to make sure that the arguments passed in are in the combined set of `formals()` of `glmnet` and `cv.glmnet`.

yes, that's another good catch, we should definitely add information to the `NEWS.md` file to describe whatever the new changes are for v0.4.3 of the package. also, if there are any other comments to discuss from the review (e.g., `ids` in the constructor versus a helper function to create folds to pass in to `fit_control` in the repeated measures case), we can migrate these to issues so that we can get this merged

**Pull request merged**